include/itkCoocurrenceTextureFeaturesImageFilter.hxx

```diff
@@ -236,7 +236,7 @@ CoocurrenceTextureFeaturesImageFilter<TInputImage, TOutputImage, TMaskImage>
 
           // Increase the corresponding bin in the histogram
           ++totalNumberOfFreq;
-          ++hist[currentInNeighborhoodPixelIntensity][pixelIntensity];
+          ++hist[(int)currentInNeighborhoodPixelIntensity][(int)pixelIntensity];
```

Comment:
I'd recommend adding a protected class typedef "HistogramIndexType" in stead of using these int types through out the code.

Reply:
floating points are just being casted, this may not give consistent across platform behavior. Consider using a rounding method from itk::Math.

What happened to floating point errors (Nan,Inf)?

Are out of bounds number guarded?